src/bind.py

```diff
@@ -196,37 +239,89 @@ def handle_relation_data(
+                shutil.move(
+                    pathlib.Path(tempdir, file_name),
+                    pathlib.Path(constants.DNS_CONFIG_DIR, file_name),
+                )
+
         # Reload charmed-bind config
         self.reload()
 
-        # Remove staging area
-        shutil.rmtree(tempdir)
+        # Return the provider data with the entries' status
+        return self._create_dns_record_provider_data(relation_data)
+
+    def _create_dns_record_provider_data(
+        self,
+        relation_data: typing.List[typing.Tuple[DNSRecordRequirerData, DNSRecordProviderData]],
+    ) -> DNSRecordProviderData:
+        """Create dns record provider data from relation data.
+
+        Args:
+            relation_data: input relation data
 
-        # Create output statuses
+        Returns:
+            A DNSRecordProviderData object with requests' status
+        """
+        zones: typing.List[Zone] = []
+        for record_requirer_data, _ in relation_data:
+            zones.extend(self._record_requirer_data_to_zones(record_requirer_data))
+        nonconflicting, conflicting = self._get_conflicts(zones)
         statuses = []
         for record_requirer_data, _ in relation_data:
-            for entry in record_requirer_data.dns_entries:
-                statuses.append(DNSProviderData(uuid=entry.uuid, status=Status.APPROVED))
+            for requirer_entry in record_requirer_data.dns_entries:
+                dns_entry = create_dns_entry_from_requirer_entry(requirer_entry)
+                if dns_entry in nonconflicting:
+                    statuses.append(
+                        DNSProviderData(uuid=requirer_entry.uuid, status=Status.APPROVED)
+                    )
+                    continue
+                if dns_entry in conflicting:
+                    statuses.append(
+                        DNSProviderData(uuid=requirer_entry.uuid, status=Status.CONFLICT)
+                    )
+                    continue
+                statuses.append(DNSProviderData(uuid=requirer_entry.uuid, status=Status.UNKNOWN))
         return DNSRecordProviderData(dns_entries=statuses)
+
+    def collect_status(
+        self,
+        event: ops.CollectStatusEvent,
+        relation_data: typing.List[typing.Tuple[DNSRecordRequirerData, DNSRecordProviderData]],
+    ) -> None:
+        """Add status for the charm based on the status of the dns record requests.
+
+        Args:
+            event: Event triggering the collect-status hook
+            relation_data: data coming from the relation databag
+        """
+        zones: typing.List[Zone] = []
+        for record_requirer_data, _ in relation_data:
+            zones.extend(self._record_requirer_data_to_zones(record_requirer_data))
+        _, conflicting = self._get_conflicts(zones)
+        if len(conflicting) > 0:
+            event.add_status(ops.BlockedStatus("Conflicting requests"))
+        event.add_status(ops.ActiveStatus())
```

Comment:
This will make all the `self.unit.status = ops.ActiveStatus()` or any other "last setting of the imot status" in the charm obsolete

Reply:
I forgot to remove them, thanks